src/core/rules.rkt

```diff
@@ -101,6 +102,21 @@
                      (filter rule-ops-supported? rules))
             (for ([rule (in-list rules)])
               (sow rule))))))
+
+;; Spec contains no accelerators
+(define (spec-has-accelerator? spec)
+  (match spec
+    [(list (? operator-accelerator?) _ ...) #t]
+    [(list _ args ...) (ormap spec-has-accelerator? args)]
+    [_ #f]))
+
+(define (real-rules rules)
```

Comment:
So AFAICT this is only called on `(*rules*)` and `(*simplify-rules*)` and only as a way to... what? Remove fp-safe rules? We can probably eventually delete this. Different PR of course.

Reply:
Filter boolean rules? This was relevant when we had the old egg IR encoding, but I'm unsure if this is still the case